x-pack/legacy/plugins/siem/index.ts

```diff
@@ -125,17 +123,38 @@ export const siem = (kibana: any) => {
       mappings: savedObjectMappings,
     },
     init(server: Server) {
-      const newPlatform = ((server as unknown) as KbnServer).newPlatform;
-      if (server.plugins.alerting != null) {
-        const type = signalsAlertType({
-          logger: newPlatform.coreContext.logger.get('plugins', APP_ID),
-        });
-        if (isAlertExecutor(type)) {
-          server.plugins.alerting.setup.registerType(type);
-        }
-      }
-      server.injectUiAppVars('siem', async () => server.getInjectedUiAppVars('kibana'));
-      initServerWithKibana(server);
+      const {
+        config,
+        getInjectedUiAppVars,
+        indexPatternsServiceFactory,
+        injectUiAppVars,
+        newPlatform,
+        plugins,
+        register,
+        route,
+        savedObjects,
+      } = server;
+
+      const { logger } = newPlatform.coreContext;
+
+      const legacy = {
+        config,
+        getInjectedUiAppVars,
+        indexPatternsServiceFactory,
+        injectUiAppVars,
+        newPlatform: {
+          coreContext: { logger },
+          env: newPlatform.env,
+        },
+        plugins: { alerting: plugins.alerting, xpack_main: plugins.xpack_main },
+        register: register.bind(server),
```

Comment:
There was an obvious failure with `register` when it was unbound, but everything else seems* to work. Should I leave the other functions unbound, or bind them just in case?

\* I should not be the person making that determination :)